src/main/java/org/terasology/core/world/generator/facetProviders/PlateauProvider.java

```diff
@@ -59,9 +59,9 @@ public PlateauProvider(Vector2ic center, float targetHeight, float innerRadius,
     @Override
     public void process(GeneratingRegion region) {
         BlockRegion reg = region.getRegion();
-        Rect2i rc = Rect2i.createFromMinAndMax(reg.minX(), reg.minZ(), reg.maxX(), reg.maxZ());
+        Rectanglei rc = new Rectanglei(reg.minX(), reg.minZ(), reg.maxX(), reg.maxZ());
```

Comment:
why does this need to be cast to Rectanglei? when you can use a BlockArea?

Reply:
Well, the reason why this is/was a Rectangle is that `distanceSquared` was implemented for `Rect2i` - and that's the only use case for the rectangle here, really...

I'm wondering whether we could get rid of the rectangle altogether and just compare the distance between
```java
if (centerPos.distanceSquared(reg.getCenterX(), reg.getCenterY()) <= outerRadius * outerRadius) {
```

However, we'd need to account for the width/height in that case...

Reply:
Another option would be to add `BlockArea#distanceSquared(x, y)` and `BlockRegion#distanceSquared(x,y,z)` accordingly and use that instead.

Reply:
sounds like something we might want to look into adding. hmm

Reply:
Yes. But let's get this merged so that we get over with JOML eventually. We still have the big module overhaul on our list, and I'd see this as a potential action item then.